### PR TITLE
Delete Mismatching Sections by Data Device

### PR DESCRIPTION
**Pull request by JingleH**
Resolves: https://jira.corp.adobe.com/browse/MWPW-133990?filter=381833

We should no longer see sections labelled with data-audience=mobile on desktop and data-audience=desktop on mobile.

Note: I attempted updating the decorateSections() to do the removal there, but we will have some mysterious padding that I suspect caused by splitSections(). So I did it the fast way, but in the future we should look into decoupling them and removing sections before even decorating them for performance enhancement.

Test URLs:
- Before: https://stage--express--adobecom.hlx.page/express/templates/banner/gaming?lighthouse=on
- After: https://del-secs-by-device--express--adobecom.hlx.page/express/templates/banner/gaming?lighthouse=on

Another pair of pages to see the difference:
- Before: https://stage--express--adobecom.hlx.page/express/create/chart?lighthouse=on
- After: https://del-secs-by-device--express--adobecom.hlx.page/express/create/chart?lighthouse=on

### CODE REVIEW

**Review by vhargrave — COMMENTED** [review]
could we actually do this like you originally planned Jingle? 
I much more prefer that approach and there's no reason to stress with this ticket.

**Comment by vhargrave** [comment]
> could we actually do this like you originally planned Jingle? I much more prefer that approach and there's no reason to stress with this ticket.

I'd actually even remove the sections before splitting the sections happens

**Comment by JingleH** [comment]
> > could we actually do this like you originally planned Jingle? I much more prefer that approach and there's no reason to stress with this ticket.
> 
> I'd actually even remove the sections before splitting the sections happens

Yeah it looks like also autoBlocking and fragment is causing some trouble... I'll need to untangle quite a few things now 🥲

But I'm glad that you think that we should do it like my original plan and strive for excellency! I'll rearrange my ticket prioritization a bit and continue working on this! I'll let you folks know when it's done

File: express/scripts/scripts.js [hunk 1]
```diff
@@ -1848,26 +1860,12 @@ async function buildAutoBlocks($main) {
...
-
-        [desktopButton, mobileButton].forEach((button) => {
-          if (button) {
-            button.classList.add('spreadsheet-powered');
-            if ($lastDiv) {
-              $lastDiv.append(button);
-            }
-          }
-        });
+      const device = document.body.dataset?.device;
```
Comment (JingleH) [inline express/scripts/scripts.js:1863]: For your context, we used to render both CTAs on the page. I am trying to only load one

File: express/scripts/scripts.js [hunk 2]
```diff
@@ -465,11 +465,23 @@ export function readBlockConfig($block) {
...
+    }
+  });
+}
+
 /**
  * Decorates all sections in a container element.
  * @param {Element} $main The container element
  */
-export function decorateSections($main) {
+function decorateSections($main) {
```
Comment (JingleH) [inline express/scripts/scripts.js:484]: For your context, these functions are never imported elsewhere. So why not save some bytes here

File: express/scripts/scripts.js [hunk 3]
```diff
@@ -1916,15 +1914,15 @@ function setTheme() {
     // mega nav, suppress brand header
     theme = 'no-brand-header';
   }
-  const $body = document.body;
+  const { body } = document;
```
Comment (JingleH) [inline express/scripts/scripts.js:1917]: Removing dollar sign while I'm here...

File: express/scripts/scripts.js [hunk 4]
```diff
@@ -2320,9 +2319,8 @@ function decorateLegalCopy(main) {
 /**
  * loads everything needed to get to LCP.
  */
-async function loadEager() {
+async function loadEager(main) {
```
Comment (JingleH) [inline express/scripts/scripts.js:2322]: like in above and below, passing main around save more bytes and allow us to use less querySelector

File: express/scripts/scripts.js [hunk 5]
```diff
@@ -2353,13 +2351,9 @@ async function loadEager() {
...
 
     const lcpBlocks = ['columns', 'hero-animation', 'hero-3d', 'template-list', 'template-x', 'floating-button', 'fullscreen-marquee', 'fullscreen-marquee-desktop', 'collapsible-card', 'search-marquee'];
-    const blocks = document.querySelectorAll('.block');
-    const firstVisualBlock = Array.from(blocks).find((b) => {
-      const { audience } = b.closest('.section')?.dataset || {};
-      return audience === document.body.dataset.device;
-    });
-    const hasLCPBlock = (firstVisualBlock && lcpBlocks.includes(firstVisualBlock.getAttribute('data-block-name')));
-    if (hasLCPBlock) await loadBlock(firstVisualBlock, true);
+    const block = document.querySelector('.block');
```
Comment (JingleH) [inline express/scripts/scripts.js:2354]: After removing irrelevant sections, our first block should usually be the LCP block now

File: express/scripts/scripts.js [hunk 6]
```diff
@@ -2482,8 +2474,9 @@ async function decoratePage() {
   window.hlx.lighthouse = new URLSearchParams(window.location.search).get('lighthouse') === 'on';
   window.hlx.init = true;
 
-  await loadEager();
-  loadLazy();
+  const main = document.querySelector('main');
```
Comment (JingleH) [inline express/scripts/scripts.js:2477]: like in above, passing main around save more bytes and allow us to use less querySelector

**Comment by JingleH** [comment]
@vhargrave I've done it the better way now. While I'm at scripts.js already, next week I'll start working on reducing its size

**Review by apganapa — APPROVED** [review]

**Review by qiyundai — APPROVED** [review]
lgtm!

**Pull request merged**